### PR TITLE
refactor and optimize code

### PR DESCRIPTION
**Pull request by midzer**
Hi all,
it's been a rather boring afternoon, so I had some time to review latest code base. I've found several opportunities to refactor and (hopefully) improve code:

* don't initialize global variables when they receive a proper value in any case <- **reverted this one**
* substitute `getElementsByTagName()[0]` to modern `querySelector`
* slightly reorder `lightbox` childs to allow removal of `focusableEls` variable
* improve `checkSupport` and introduce `checkType` helper function for non-image elements
* prefer conditional operator
* prefer `textContent` over `innerHTML` where it makes sense https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent#Differences_from_innerHTML and https://html.com/attributes/img-alt/
* rename `x` to `figcaptionId`
* prefer shorter null checks like `if (video)`
* check for `hasOwnProperty()` in for-in loops (via https://leftshift.io/4-javascript-optimisations-you-should-know)
* variable `elements` only needed locally
* **UPDATE**: introduce `reset()` for setting lightbox to original state (useful in combination with dynamic `add()`ing of elements

Looking forward to any feedback about this PR.

Have a nice weekend
midzer

edit:
those changes squeezed out about 20ms scripting time over last version on my (rather old) laptop for adding 48 elements (images and videos) dynamically

### CODE REVIEW

**Comment by midzer** [comment]
Hi again.
Been a lazy sunday. So had I had time to look over your changes as well. Merge as been successful and looking good so for, have it running in production environment :)

I've updated initial post to reflect whats been changed (or reverted).

File: js/tobi.js [hunk 1]
```diff
@@ -502,9 +458,8 @@
       // Save the user’s focus
       lastFocus = document.activeElement
 
-      focusableEls = lightbox.querySelectorAll('button')
-      firstFocusableEl = focusableEls[0]
-      lastFocusableEl = focusableEls[focusableEls.length - 1]
+      firstFocusableEl = overlay.firstElementChild
```
Comment (rqrauhvmra) [inline js/tobi.js]: `overlay` has no child elements

File: js/tobi.js [hunk 2]
```diff
@@ -502,9 +458,8 @@
...
+      lastFocusableEl = overlay.lastElementChild
```
Comment (rqrauhvmra) [inline js/tobi.js]: `overlay` has no child elements

File: js/tobi.js [hunk 3]
```diff
@@ -458,8 +458,8 @@
       // Save the user’s focus
       lastFocus = document.activeElement
 
-      firstFocusableEl = overlay.firstElementChild
-      lastFocusableEl = overlay.lastElementChild
+      firstFocusableEl = lightbox.firstElementChild
```
Comment (rqrauhvmra) [inline js/tobi.js]: `lightbox.firstElementChild` is `div.tobi__overlay`.

Regarding the restructuring I used the solution with `focusableEls = lightbox.querySelectorAll ('button')`.

Comment (midzer) [reply js/tobi.js]: Check current code. I reshuffled ìtems which are appended to `lightbox`. `lightbox.appendChild(prevButton)` comes first.

Comment (rqrauhvmra) [reply js/tobi.js]: Sorry, I overlooked that.

Comment (rqrauhvmra) [reply js/tobi.js]: I will take it over but still test it. I still have to catch a few cases:
If it is just an element and later more or if the navigation buttons / close button is disabled ...

**Pull request merged**